README.md

```diff
@@ -11,4 +11,237 @@
 [![Maven Central](https://img.shields.io/maven-central/v/com.artipie/npm-proxy-adapter.svg)](https://maven-badges.herokuapp.com/maven-central/com.artipie/npm-proxy-adapter)
 [![PDD status](http://www.0pdd.com/svg?name=yegor256/npm-files)](http://www.0pdd.com/p?name=yegor256/npm-files)
 
-TBD...
+# npm-proxy-adapter
+
+## Usage
+This is the dependency you need:
+
+```xml
+<dependency>
+  <groupId>com.artipie</groupId>
+  <artifactId>npm-proxy-adapter</artifactId>
+  <version>[...]</version>
+</dependency>
+```
+
+## How it works?
+
+After initial setup of the proxy repository, it can be used to work with NPM.
+NPM registry to install package from can be specified explicitly 
+`npm install --registry=https://artipie.com/npm-proxy-registry <package_name>` or setup 
+as default with `npm set registry https://artipie.com/npm-proxy-registry` command. 
+NPM sends HTTP requests to specified URL and adapter processes them. It proxies 
+request to the remote repository previously setup. After getting artifact, it caches 
+received data and transform before sending it back to client. The next request will not 
+require remote repository call - the cached data will be used.
+
+## Technical details
+
+[Public NPM registry API ](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md) 
+describes 6 methods. And one more method is called by `npm` client to get
+security audit information. We will support just one in the beginning: get package by name.
+
+### Get package by name method description
+Receive request from `npm` client. `{package}` parameter can be in simple (`package-name`) or 
+scoped (`@scope/package-name`) form:
+```http request
+GET /path-to-repository/{package}
+connection: keep-alive
+user-agent: npm/6.14.3 node/v10.15.2 linux x64
+npm-in-ci: false
+npm-scope: 
+npm-session: 439998791e27a7b1
+referer: install
+pacote-req-type: packument
+pacote-pkg-id: registry:minimalistic-assert
+accept: application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*
+accept-encoding: gzip,deflate
+Host: artipie.com
+```
+At first, we try to get `{package}/package.json` file in the storage. If there is no
+such file, we send request from `npm-proxy-adapter` to remote registry:
+```http request
+GET /{package} HTTP/1.1
+Host: registry.npmjs.org
+Connection: Keep-Alive
+User-Agent: Artipie/1.0.0
+Accept-Encoding: gzip,deflate
+```
+Than we handle response from remote registry. Process headers:
+```http request
```

Comment:
Those dumps turn the readme into an unreadable mess. Avoid it

Reply:
It's because of this file was initially design document. I have added issue to rewrite it. https://github.com/artipie/npm-proxy-adapter/issues/5